WordPress/gutenberg
# Core Data: Add docgen auto-documentation for actions, selectors

This pull request seeks to update the docgen auto-documentation script to generate documentation for actions and selectors of the `@wordpress/core-data` module.

**Implementation notes:**

I could see this being iterated upon toward:

- Eliminating the separate [`docs/tool` script](https://github.com/WordPress/gutenberg/tree/master/docs/tool) in favor of consolidating all data documentation here
   - Mentioned in "Implementation notes" of #15188
- Creating a custom formatter for actions / selectors to make them more accurate / relevant
   - Omit actions `return`
   - Omit selectors `state` argument
- Provide examples for using various `core-data` selectors
   - See #12120

**Testing Instructions:**

Verify there are no local changes after running `npm run docs:build` in this branch.
